keyvi/include/keyvi/vector/vector_generator.h

```diff
@@ -37,10 +37,11 @@
 namespace keyvi {
 namespace vector {
 
-template <class ValueStoreT>
+template <keyvi::dictionary::fsa::internal::value_store_t DictionaryType>
```

Comment:
sounds more like a naming issue to me: I haven't thought about vector when calling the type `Dict<...>` - thinking about it, it would make sense to rename. All this does is simplifying the selection of a value store. 

Any suggestions for a better name?

Reply:
Actually I think that this class should have a member for value store reader as well, which will ultimately eliminate a need of having `ValueStoreFactory` as a separate class/method. Also it will ensure that someone implementing new value store, implements all it's components.

So maybe `value_store_components` ?

Reply:
sounds good, I doubt we can get rid of the factory as this is part of runtime abstraction but I think we can do some improvements